src/main/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheck.java

```diff
@@ -156,6 +164,15 @@ private boolean isMatch(int lineNumber, String line) {
             || getHeaderLines().get(lineNumber).equals(line);
     }
 
+    /**
+     * Checks if the file is set to be skipped.
+     * @param file the file to check whether it should be skipped.
+     * @return true if and only if the file name matches the skipFiles regex string.
+     */
+    private boolean skipFile(File file) {
+        return skipFiles.matcher(file.getName()).find();
```

Comment:
please review https://github.com/checkstyle/checkstyle/issues/2986#issuecomment-605454689

you code here will check for filename only (without path), ,filter will do suppression by whole path (more powerful solution). We need here absolute path to be used, to let users skip violations in certain packages only.
right now we are in